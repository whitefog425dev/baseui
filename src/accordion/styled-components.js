/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import {styled} from '../styles/index.js';
import {getSvgStyles} from '../icon/styled-components.js';
import type {SharedStylePropsT} from './types.js';

/**
 * Main component container element
 */
export const Root = styled('div', (props: SharedStylePropsT) => {
  return {};
});

export const PanelContainer = styled('div', (props: SharedStylePropsT) => {
  return {width: '100%'};
});

export const Header = styled('h3', (props: SharedStylePropsT) => {
  const {
    $disabled,
    $expanded,
    $theme: {colors, sizing, typography},
  } = props;
  return {
    ...typography.font450,
    color: colors.foreground,
    cursor: $disabled ? 'not-allowed' : 'pointer',
    background: colors.listHeaderFill,
    paddingTop: sizing.scale600,
    paddingBottom: sizing.scale600,
    paddingLeft: sizing.scale700,
    paddingRight: sizing.scale1200,
    marginTop: 0,
    marginBottom: 0,
    marginLeft: 0,
    marginRight: 0,
    borderBottom: `1px solid ${$expanded ? colors.mono500 : colors.mono400}`,
    position: 'relative',
    display: 'flex',
    alignItems: 'center',
    ':hover': {
      color: colors.linkHover,
    },
    ':focus': {
      backgroudColor: colors.background,
    },
  };
});

export const ToggleIcon = styled('svg', props => {
  const {$theme, $disabled, $color} = props;
  return {
    ...getSvgStyles(props),
    color: $color || $theme.colors.foreground,
    cursor: $disabled ? 'not-allowed' : 'pointer',
    position: 'absolute',
    right: '12px',
  };
});

export const Content = styled('div', (props: SharedStylePropsT) => {
  const {
    $theme: {animation, colors, sizing, typography},
    $expanded,
  } = props;
  return {
    ...typography.font300,
    backgroundColor: colors.background,
    color: colors.foreground,
    paddingTop: $expanded ? sizing.scale800 : 0,
    paddingBottom: $expanded ? sizing.scale1000 : 0,
    paddingLeft: sizing.scale800,
    paddingRight: sizing.scale800,
    marginTop: 0,
    marginBottom: 0,
    marginLeft: 0,
    marginRight: 0,
    borderBottom: $expanded ? `1px solid ${colors.border}` : 'none',
    boxSizing: 'border-box',
    height: $expanded ? 'auto' : '0',
    maxHeight: $expanded ? '100%' : '0',
    overflow: 'hidden',
    opacity: $expanded ? 1 : 0,
    transitionProperty: 'all',
    transitionDuration: animation.timing400,
    transitionTimingFunction: animation.easeInOutCurve,
  };
});
