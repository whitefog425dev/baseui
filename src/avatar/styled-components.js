/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import {styled} from '../styles/index.js';
import type {StylePropsT} from './types.js';

function getSize(props: StylePropsT) {
  const {$size, $theme} = props;

  const defaultSize = $theme.sizing.scale1000;
  const size = $size || defaultSize;
  return $theme.sizing[size] || size;
}

export const Avatar = styled('img', (props: StylePropsT) => {
  const themedSize = getSize(props);

  return {
    borderRadius: '50%',
    boxSizing: 'border-box',
    display: 'block',
    height: themedSize,
    width: themedSize,
  };
});
Avatar.displayName = 'StyledAvatar';

export const Root = styled('div', (props: StylePropsT) => {
  const {$didImageFailToLoad} = props;
  const themedSize = getSize(props);

  return {
    backgroundColor: props.$theme.colors.mono400,
    borderRadius: '50%',
    boxSizing: 'border-box',
    display: 'inline-block',

    // image previously set the root height/width
    // since image is not rendered, set the height/width
    height: $didImageFailToLoad ? themedSize : null,
    width: $didImageFailToLoad ? themedSize : null,
  };
});
Root.displayName = 'StyledRoot';
