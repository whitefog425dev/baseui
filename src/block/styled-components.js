/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

// @flow

import {styled} from '../styles/index.js';
import type {Font} from '../styles/types.js';
import type {StyledBlockPropsT} from './types.js';

export const StyledBlock = styled(
  'div',
  ({
    $theme,
    $color,
    $font,
    $alignContent,
    $alignItems,
    $alignSelf,
    $flexDirection,
    $display,
    $flex,
    $justifyContent,
    $justifySelf,
    $position,
    $width,
    $minWidth,
    $maxWidth,
    $height,
    $minHeight,
    $maxHeight,
    $overflow,
    $margin,
    $marginTop,
    $marginRight,
    $marginBottom,
    $marginLeft,
    $padding,
    $paddingTop,
    $paddingRight,
    $paddingBottom,
    $paddingLeft,
    $flexWrap,
    $left,
    $top,
    $right,
    $bottom,
    ...other
  }: StyledBlockPropsT) => {
    const {colors, typography, sizing} = $theme;

    let styles = {};

    if ($color !== undefined) {
      styles.color = colors[$color] || $color;
    }

    if ($font !== undefined) {
      const font: Font = typography[$font];
      styles.fontFamily = font.fontFamily;
      styles.fontWeight = font.fontWeight;
      styles.fontSize = font.fontSize;
      styles.lineHeight = font.lineHeight;
    }

    if ($alignContent !== undefined) {
      styles.alignContent = $alignContent;
    }

    if ($alignItems !== undefined) {
      styles.alignItems = $alignItems;
    }

    if ($alignSelf !== undefined) {
      styles.alignSelf = $alignSelf;
    }

    if ($flexDirection !== undefined) {
      styles.flexDirection = $flexDirection;
    }

    if ($display !== undefined) {
      styles.display = $display;
    }

    if ($flex !== undefined) {
      styles.flex = $flex;
    }

    if ($justifyContent !== undefined) {
      styles.justifyContent = $justifyContent;
    }

    if ($justifySelf !== undefined) {
      styles.justifySelf = $justifySelf;
    }

    if ($position !== undefined) {
      styles.position = $position;
    }

    if ($width !== undefined) {
      styles.width = $width;
    }

    if ($minWidth !== undefined) {
      styles.minWidth = $minWidth;
    }

    if ($maxWidth !== undefined) {
      styles.maxWidth = $maxWidth;
    }

    if ($height !== undefined) {
      styles.height = $height;
    }

    if ($minHeight !== undefined) {
      styles.minHeight = $minHeight;
    }

    if ($maxHeight !== undefined) {
      styles.maxHeight = $maxHeight;
    }

    if ($overflow === 'scrollX') {
      styles.overflowX = 'scroll';
    } else if ($overflow === 'scrollY') {
      styles.overflowY = 'scroll';
    } else if ($overflow !== undefined) {
      styles.overflow = $overflow;
    }

    if ($margin !== undefined) {
      styles.margin = sizing[$margin] || $margin;
    }

    if ($marginTop !== undefined) {
      styles.marginTop = sizing[$marginTop] || $marginTop;
    }

    if ($marginRight !== undefined) {
      styles.marginRight = sizing[$marginRight] || $marginRight;
    }

    if ($marginBottom !== undefined) {
      styles.marginBottom = sizing[$marginBottom] || $marginBottom;
    }

    if ($marginLeft !== undefined) {
      styles.marginLeft = sizing[$marginLeft] || $marginLeft;
    }

    if ($padding !== undefined) {
      styles.padding = sizing[$padding] || $padding;
    }

    if ($paddingTop !== undefined) {
      styles.paddingTop = sizing[$paddingTop] || $paddingTop;
    }

    if ($paddingRight !== undefined) {
      styles.paddingRight = sizing[$paddingRight] || $paddingRight;
    }

    if ($paddingBottom !== undefined) {
      styles.paddingBottom = sizing[$paddingBottom] || $paddingBottom;
    }

    if ($paddingLeft !== undefined) {
      styles.paddingLeft = sizing[$paddingLeft] || $paddingLeft;
    }

    if ($flexWrap !== undefined) {
      styles.flexWrap = 'wrap';
    }

    if ($left !== undefined) {
      styles.left = $left;
    }

    if ($top !== undefined) {
      styles.top = $top;
    }

    if ($right !== undefined) {
      styles.right = $right;
    }

    if ($bottom !== undefined) {
      styles.bottom = $bottom;
    }

    return styles;
  },
);
