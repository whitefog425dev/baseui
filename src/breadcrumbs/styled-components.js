/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

// @flow

import {styled} from '../styles/index.js';
import type {StyledRootPropsT, StyledSeparatorT} from './types.js';
import {ChevronRight} from '../icon/index.js';

export const StyledRoot = styled('nav', ({$theme}: StyledRootPropsT) => {
  return {
    color: $theme.colors.breadcrumbsText,
    ...$theme.typography.font450,
  };
});
StyledRoot.displayName = 'StyledRoot';

export const StyledSeparator = styled('div', ({$theme}: StyledSeparatorT) => {
  return {
    display: 'inline-block',
    color: $theme.colors.breadcrumbsSeparatorFill,
    marginLeft: $theme.sizing.scale300,
    marginRight: $theme.sizing.scale300,
  };
});
StyledSeparator.displayName = 'StyledSeparator';

export const StyledIcon = styled(ChevronRight, () => {
  return {
    verticalAlign: 'text-bottom',
  };
});
StyledIcon.displayName = 'StyledIcon';
