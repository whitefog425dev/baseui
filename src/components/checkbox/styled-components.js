// @flow
import {styled} from '../../styles';

function getBorderColor(props) {
  const {$disabled, $checked, $isError, $isIndeterminate, $theme} = props;
  const {colors} = $theme;
  if ($disabled) {
    return colors.mono300;
  } else if ($checked || $isIndeterminate) {
    return 'transparent';
  } else if ($isError) {
    return colors.negative400;
  } else {
    return colors.mono700;
  }
}

function getLabelPadding(props) {
  const {$labelPlacement = '', $theme} = props;
  const {sizing} = $theme;
  const {scale200} = sizing;
  const paddingSide =
    'padding' + $labelPlacement.replace(/^\w/, c => c.toUpperCase());
  return {
    [paddingSide]: scale200,
  };
}

function getBackgroundColor(props) {
  const {
    $disabled,
    $checked,
    $isIndeterminate,
    $isFocused,
    $isError,
    $isHovered,
    $isActive,
    $theme,
  } = props;
  const {colors} = $theme;
  if ($disabled) {
    return colors.mono300;
  } else if ($isError && ($isIndeterminate || $checked)) {
    if ($isActive || $isFocused) {
      return colors.negative600;
    } else if ($isHovered) {
      return colors.negative500;
    } else {
      return colors.negative400;
    }
  } else if ($isError) {
    if ($isActive || $isFocused) {
      return colors.negative200;
    } else if ($isHovered) {
      return colors.negative100;
    } else {
      return colors.negative50;
    }
  } else if ($isIndeterminate || $checked) {
    if ($isActive || $isFocused) {
      return colors.primary600;
    } else if ($isHovered) {
      return colors.primary500;
    } else {
      return colors.primary400;
    }
  } else {
    if ($isActive || $isFocused) {
      return colors.mono500;
    } else if ($isHovered) {
      return colors.mono400;
    } else {
      return 'transparent';
    }
  }
}

function getCheckBackgroundColor(props) {
  const {$disabled, $theme} = props;
  const {colors} = $theme;
  return $disabled ? colors.mono600 : colors.mono100;
}

function getLabelColor(props) {
  const {$disabled, $theme} = props;
  const {colors} = $theme;
  return $disabled ? colors.mono600 : colors.mono1000;
}

export const Root = styled('label', props => {
  const {$disabled, $labelPlacement} = props;
  return {
    flexDirection:
      $labelPlacement === 'top' || $labelPlacement === 'bottom'
        ? 'column'
        : 'row',
    display: 'flex',
    alignItems:
      $labelPlacement === 'top' || $labelPlacement === 'bottom'
        ? 'center'
        : 'flex-start',
    cursor: $disabled ? 'not-allowed' : 'pointer',
    userSelect: 'none',
  };
});

export const Checkmark = styled('span', props => {
  const {$checked, $isIndeterminate, $theme} = props;
  const {sizing, animation} = $theme;
  return {
    flex: '0 0 auto',
    transitionDuration: animation.timing100,
    transitionTimingFunction: animation.easeOutCurve,
    width: sizing.scale600,
    height: sizing.scale600,
    left: '4px',
    top: '4px',
    borderStyle: 'solid',
    borderWidth: '2px',
    borderColor: getBorderColor(props),
    borderRadius: $theme.borders.useRoundedCorners
      ? $theme.borders.radius200
      : null,
    display: 'inline-block',
    verticalAlign: 'middle',
    backgroundImage: $isIndeterminate
      ? `url('data:image/svg+xml;utf8,<svg width="12" height="2" viewBox="0 0 12 2" fill="none" xmlns="http://www.w3.org/2000/svg"><line x1="1" y1="-1" x2="11" y2="-1" transform="translate(0 2)" stroke="${getCheckBackgroundColor(
          props,
        )}" stroke-width="2" stroke-linecap="round"/></svg>');`
      : $checked
        ? `url('data:image/svg+xml;utf8,<svg width="11" height="10" viewBox="0 0 11 10" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M10.6 0.200059C11.0418 0.53143 11.1314 1.15823 10.8 1.60006L4.8 9.60006C4.62607 9.83197 4.36005 9.97699 4.07089 9.99754C3.78173 10.0181 3.49788 9.91215 3.29289 9.70717L0.292893 6.70717C-0.0976311 6.31664 -0.0976311 5.68348 0.292893 5.29295C0.683417 4.90243 1.31658 4.90243 1.70711 5.29295L3.89181 7.47765L9.2 0.400059C9.53137 -0.0417689 10.1582 -0.131312 10.6 0.200059Z" fill="${getCheckBackgroundColor(
            props,
          )}"/></svg>');`
        : null,
    backgroundColor: getBackgroundColor(props),
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    marginTop: $theme.sizing.scale0,
    marginBottom: $theme.sizing.scale0,
    marginLeft: $theme.sizing.scale200,
    marginRight: $theme.sizing.scale200,
  };
});

export const Label = styled('div', props => {
  const {$theme} = props;
  const {typography} = $theme;
  return {
    verticalAlign: 'middle',
    ...getLabelPadding(props),
    color: getLabelColor(props),
    ...typography.font350,
  };
});
// tricky style for focus event cause display: none doesn't work
export const Input = styled('input', {
  opacity: 0,
  width: 0,
  height: 0,
  overflow: 'hidden',
  margin: 0,
  padding: 0,
});
