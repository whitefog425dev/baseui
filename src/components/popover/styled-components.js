// @flow
import {styled} from '../../styles';
import {ARROW_WIDTH} from './constants';
import {
  getPopoverMarginStyles,
  getArrowPositionStyles,
  getStartPosition,
  getEndPosition,
} from './utils';
import type {SharedStylePropsT} from './types';

/**
 * Main popover container element that gets positioned next to the anchor
 */
export function getBodyStyles(props: SharedStylePropsT) {
  const {
    $isOpen,
    $isAnimating,
    $placement,
    $popoverOffset,
    $showArrow,
    $theme,
  } = props;

  return {
    position: 'absolute',
    top: 0,
    left: 0,
    zIndex: 1050,
    backgroundColor: $theme.colors.background,
    borderRadius: $theme.borders.useRoundedCorners
      ? $theme.borders.radius300
      : '0px',
    boxShadow: $theme.lighting.shadow600,

    transitionProperty: 'opacity,transform',
    transitionDuration: $isAnimating ? '0.1s' : '0s',
    transitionTimingFunction: $isOpen
      ? $theme.animation.easeOutCurve
      : $theme.animation.easeInCurve,
    opacity: $isAnimating && $isOpen ? 1 : 0,
    transform:
      $isAnimating && $isOpen
        ? getEndPosition($popoverOffset)
        : getStartPosition($popoverOffset, $placement, $showArrow),
    ...getPopoverMarginStyles($showArrow, $placement),
  };
}

export const Body = styled('div', getBodyStyles);

/**
 * Arrow shown between the popover and the anchor element
 */
export function getArrowStyles(props: SharedStylePropsT) {
  const {$arrowOffset, $placement, $theme} = props;
  return {
    backgroundColor: $theme.colors.background,
    boxShadow: $theme.lighting.shadow600,
    width: `${ARROW_WIDTH}px`,
    height: `${ARROW_WIDTH}px`,
    transform: 'rotate(45deg)',
    position: 'absolute',
    zIndex: 1, // Below "Inner"
    ...getArrowPositionStyles($arrowOffset, $placement),
  };
}

export const Arrow = styled('div', getArrowStyles);

/**
 * Extra div that holds the popover content. This extra element
 * is needed for the arrowâ€“the arrow is just a 45deg rotated div,
 * and rendering this extra element on top with a solid background
 * clips the part of the arrow that extends into the popover.
 */
export function getInnerStyles({$theme}: SharedStylePropsT) {
  return {
    backgroundColor: $theme.colors.background,
    borderRadius: $theme.borders.useRoundedCorners
      ? $theme.borders.radius300
      : '0px',
    position: 'relative',
    zIndex: 2, // Above "Arrow"
  };
}

export const Inner = styled('div', getInnerStyles);

/**
 * A drop-in component that provides the recommended padding
 * for popovers. Mostly a convenience for users so they don't
 * have to define this themselves.
 */
export const Padding = styled('div', {
  padding: '12px',
});
