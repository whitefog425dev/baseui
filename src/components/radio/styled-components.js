// @flow
import {styled} from '../../styles/index';

function getBorderColor(props) {
  const {$checked, $isError, $theme} = props;
  const {colors} = $theme;
  return $isError
    ? colors.negative400
    : $checked ? colors.primary400 : colors.mono700;
}

function getLabelPadding(props) {
  const {$labelPlacement = '', $theme} = props;
  const {sizing: {scale200}} = $theme;
  const paddingSide =
    'padding' + $labelPlacement.replace(/^\w/, c => c.toUpperCase());
  return {
    [paddingSide]: scale200,
  };
}
function getLabelColor(props) {
  const {$disabled, $theme} = props;
  const {colors} = $theme;
  return $disabled ? colors.mono600 : colors.mono1000;
}

export const RadioGroupRoot = styled('div', props => {
  const {$disabled, $align} = props;
  return {
    flexDirection: $align === 'horizontal' ? 'row' : 'column',
    display: 'flex',
    alignItems: $align === 'horizontal' ? 'center' : 'flex-start',
    cursor: $disabled ? 'not-allowed' : 'pointer',
  };
});

export const Root = styled('label', props => {
  const {$disabled, $labelPlacement} = props;
  return {
    flexDirection:
      $labelPlacement === 'top' || $labelPlacement === 'bottom'
        ? 'column'
        : 'row',
    display: 'flex',
    alignItems: 'center',
    cursor: $disabled ? 'not-allowed' : 'pointer',
  };
});

export const RadioMark = styled('span', props => {
  const {$checked, $disabled, $theme, $isFocused, $isError} = props;
  const {colors, animation} = $theme;
  const {sizing: {scale0, scale100, scale300, scale600}} = $theme;
  const activeStyle = {
    backgroundColor:
      $checked || $isError
        ? null
        : $isFocused
          ? colors.mono500
          : !$disabled && !$checked ? colors.mono400 : null,
  };
  return {
    backgroundColor: 'transparent',
    borderRadius: '50%',
    borderWidth: $checked ? scale300 : scale0,
    width: $checked ? scale100 : scale600,
    height: $checked ? scale100 : scale600,
    backgroundImage: null,
    flex: '0 0 auto',
    transitionDuration: animation.timing100,
    transitionTimingFunction: animation.easeOutCurve,
    borderStyle: 'solid',
    borderColor: getBorderColor(props),
    display: 'inline-block',
    verticalAlign: 'middle',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    marginTop: $theme.sizing.scale200,
    marginBottom: $theme.sizing.scale200,
    marginLeft: $theme.sizing.scale200,
    marginRight: $theme.sizing.scale200,
    ':hover': activeStyle,
    ':active': activeStyle,
  };
});

export const Label = styled('div', props => {
  const {$theme: {typography}} = props;
  return {
    verticalAlign: 'middle',
    ...getLabelPadding(props),
    color: getLabelColor(props),
    ...typography.font400,
  };
});
// tricky style for focus event cause display: none doesn't work
export const Input = styled('input', {
  opacity: 0,
  width: 0,
  overflow: 'hidden',
  margin: 0,
  padding: 0,
});
