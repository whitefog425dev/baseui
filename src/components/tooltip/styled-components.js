// @flow
import {styled} from '../../styles';
import {
  getBodyStyles,
  getArrowStyles,
  getInnerStyles,
} from '../popover/styled-components';
import {getEndPosition} from '../popover/utils';

/**
 * We want to reuse the default Popover styles and add our overrides
 * for Tooltip. Normally you could do something like
 * withStyleDeep(StyledPopoverBody, {...}), however this break the
 * ability to use $style properlyâ€“the style overrides passed in by
 * the customer will override the base styles but not the styles
 * specified here.
 *
 * We could create our own withStyleDeep that also takes the $style
 * prop into account
 */
export const Body = styled('div', props => ({
  ...getBodyStyles(props),
  // Tooltip
  backgroundColor: props.$theme.tooltip.backgroundColor,
  borderRadius: props.$theme.borders.radius200,
  boxShadow: props.$theme.lighting.shadow400,
  color: props.$theme.colors.mono100,
  // Reset transition property to opacity only, and static transform
  transitionProperty: 'opacity',
  transform: getEndPosition(props.$popoverOffset),
}));

export const Inner = styled('div', props => ({
  ...getInnerStyles(props),
  backgroundColor: props.$theme.tooltip.backgroundColor,
  borderRadius: props.$theme.borders.radius200,
  paddingTop: props.$theme.sizing.scale300,
  paddingBottom: props.$theme.sizing.scale300,
  paddingLeft: props.$theme.sizing.scale300,
  paddingRight: props.$theme.sizing.scale300,
  ...props.$theme.typography.font200,
}));

export const Arrow = styled('div', props => ({
  ...getArrowStyles(props),
  backgroundColor: props.$theme.tooltip.backgroundColor,
}));
