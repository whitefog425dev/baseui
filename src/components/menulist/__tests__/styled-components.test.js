// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {List, ListItem} from '../styled-components';

describe('Menulist Styled Components', () => {
  test('ListItem - basic render', () => {
    const component = shallow(<ListItem>This is a list item</ListItem>);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'ListItem has correct styles',
    );
  });

  test('ListItem - highlighted render', () => {
    const component = shallow(
      <ListItem $isHighlighted={true}>
        This is a highlighted list item
      </ListItem>,
    );
    expect(component.instance().getStyles()).toMatchSnapshot(
      'ListItem has correct styles when highlighted',
    );
  });

  test('List - basic render', () => {
    const component = shallow(<List />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'List has correct styles',
    );
  });
});
