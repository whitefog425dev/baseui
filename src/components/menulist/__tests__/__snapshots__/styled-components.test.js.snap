// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menulist Styled Components List - basic render: List has correct styles 1`] = `
Object {
  "background": "$theme.colors.white",
  "borderRadius": "$theme.borders.radius300",
  "boxShadow": "$theme.lighting.shadow600",
  "display": "flex",
  "flexDirection": "column",
  "margin": 0,
  "paddingBottom": "$theme.sizing.scale200",
  "paddingLeft": "$theme.sizing.scale400",
  "paddingRight": "$theme.sizing.scale400",
  "paddingTop": "$theme.sizing.scale200",
  "position": "relative",
}
`;

exports[`Menulist Styled Components ListItem - basic render: ListItem has correct styles 1`] = `
Object {
  ":hover": Object {
    "color": "$theme.colors.primary400",
  },
  "color": "$theme.colors.black",
  "cursor": "pointer",
  "display": "block",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "margin": 0,
  "position": "relative",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "color",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
}
`;

exports[`Menulist Styled Components ListItem - highlighted render: ListItem has correct styles when highlighted 1`] = `
Object {
  ":hover": Object {
    "color": "$theme.colors.primary400",
  },
  "color": "$theme.colors.primary",
  "cursor": "pointer",
  "display": "block",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "margin": 0,
  "position": "relative",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "color",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
}
`;
